apps/server/src/services/institutionSyncer.ts

```diff
@@ -0,0 +1,70 @@
+import { writeFileSync } from 'fs'
+import { setIntervalAsync } from 'set-interval-async'
+import config from '../config'
+import { info, warning as logWarning } from '../infra/logger'
+import {
+  deleteRemovedInstitutions,
+  getInstitutionFilePath,
+  updateInstitutions
+} from '../services/ElasticSearchClient'
+import { get, setNoExpiration } from '../services/storageClient/redis'
+import type { CachedInstitution } from '../shared/contract'
+import { getAccessToken } from './auth0Service'
+
+const RESPONSE_NOT_MODIFIED = 304
+const SUCCESS_RESPONSE = 200
+const UNAUTHORIZED_RESPONSE = 401
+const INSTITUTION_ETAG_REDIS_KEY = 'institutionEtag'
+
+export async function setInstitutionSyncSchedule(minutes: number = 5) {
+  return setIntervalAsync(
+    async () => {
+      info('Checking for institution list updates')
+      await loadCachedInstitutions()
+    },
+    minutes * 60 * 1000
+  )
+}
+
+export const loadCachedInstitutions = async () => {
+  const institutionCacheETag = await get(INSTITUTION_ETAG_REDIS_KEY)
+  try {
+    const accessToken = await getAccessToken()
+    const response = await fetch(config.InstitutionCacheListUrl, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        'If-None-Match': institutionCacheETag,
+        'Cache-Control': 'public'
+      }
+    })
+    if (response.status === RESPONSE_NOT_MODIFIED) {
+      info('Institution Cache List unchanged. Skipping Update')
+    } else if (response.status === SUCCESS_RESPONSE) {
+      info('Updating institution cache list')
+      await updateElasticSearchDocuments(response)
```

Comment:
I think we should check that there are institutions in the response before we make changes or cache the file. We could take a widget down if the institutions list came back empty.